### ISSUE TITLE
Can't login to test environment

### ISSUE THREAD

**Issue by punyflash** [telegramdesktop/tdesktop#26361]
### Steps to reproduce

1. Open `☰ Settings` > `Shift + Alt` + Right click `Add Account` and select `Test Server` ([WebApps](https://core.telegram.org/bots/webapps#testing-web-apps))
2. Click `Login using phone number`
3. Input your phone and click `Continue`
4. Telegram desktop asks you to input the code that was sent on other device (which you don't have)
5. Click `Send code using SMS`
6. Get the error that you need to create Telegram using phone

### Expected behaviour

Should be able to log in using SMS, as you can't create an account for Test Environment from mobile or may not be able to access other device.

### Actual behaviour

Can't access Test Enviromnemnt at all.

### Operating system

Windows 11

### Version of Telegram Desktop

4.8.3 x64

### Installation source

Static binary from official website

**Comment by xsr** [comment]
same, I guess mb only some countries whitelisted

**Comment by punyflash** [comment]
> same, I guess mb only some countries whitelisted

That's not the case, as I was able to login before.

**Comment by xsr** [comment]
Look I think this is not tdesktop related, I just accidentally come here from Google I guess. I will look through where also we should ask and link here. Have and idea, wait some time...

**Comment by AlexWayfer** [comment]
New details from me:

I can't login via SMS for Telegram Desktop even in production. So, it seems like a TDesktop issue. From Android I've got SMS from `VFSSMS`. And there was 6-digits code, while Telegram Desktop is expecting only 5 digits.

**Comment by john-preston** [comment]
@AlexWayfer from TDesktop you should see information that the code was sent to another app, on mobile. If you press "Send SMS" it should show you information that SMS login isn't available on TDesktop, only on the official mobile apps. After you login by SMS on the official mobile apps you can login on TDesktop using QR code or a digit-code sent to the mobile app.

This is expected behavior, server doesn't send SMS messages to TDesktop at all, only to mobile apps.

Unfortunately, currently only iOS store app can allow switching to Test environment to create an account, I'll ask about Android app to allow it somehow as well. For iOS tap quickly 10 times on the Settings icon in the bottom menu, a debug menu will open, tap "Accounts" there, tap "Login to another account" and choose "Test".

Anyway, there is nothing I can do from TDesktop side, because this is the server which doesn't allow SMS to TDesktop. We'll need to wait for Android store version or at least Android public beta version to allow switching to Test environment.

**Comment by AlexWayfer** [comment]
> If you press "Send SMS" it should show you information that SMS login isn't available on TDesktop, only on the official mobile apps.

When I've created the issue, I saw no message box about this. It behaved like SMS has been sent and expected a code. Now I see. I'll double check it later on Linux, where originally I've faced with the issue (unclear behavior).



> I'll ask about Android app to allow it somehow as well

Thanks. Yes, this is sadly, that official Android app doesn't allow you to sign-up on test servers.

I've resolved my problem with Android Beta app, but it's not available from the official sources (like website), and any download is unsafe. Anyway, I see no large danger for just test servers, without production-servers login.
